snakelings/execution.py

```diff
@@ -33,4 +35,29 @@ def execute_exercise_code(exercise: Exercise) -> Tuple[bool, str]:
 
     logger.debug(f"Return code: {return_code}")
 
-    return True if return_code == 0 else False, output if return_code == 0 else output_error
\ No newline at end of file
+    return True if return_code == 0 else False, output if return_code == 0 else output_error
+
+def execute_pytest(exercise: Exercise) -> Tuple[bool, str]:
+    main_py_path = exercise.path.joinpath("main.py").absolute()
+
+    logger.debug(f"Using pytest to execute '{main_py_path}'...")
+
+    output_buffer = io.StringIO()
+
+    sys.stdout = output_buffer
+    sys.stderr = output_buffer
+
+    return_code = pytest.main([main_py_path])
+
+    sys.stdout = sys.__stdout__
+    sys.stderr = sys.__stderr__
+
+    return True if return_code == 0 else False, "" if return_code == 0 else output_buffer.getvalue()
+
+def handle_execution(exercise: Exercise) -> Tuple[bool, str]:
+    if exercise.use_pytest is False:
+        result, output = execute_exercise_code(exercise)
+    else:
+        result, output = execute_pytest(exercise)
+    
+    return result, output
```

Comment:
This can be simplified with just one if statement and returning the value right on the spot `return execute_pytest()`.